content/blog/2019/06/2019-06-14-performance-testing-jenkins.adoc

```diff
@@ -0,0 +1,194 @@
+---
+layout: post
+title: "Microbenchmarking Framework for Jenkins Plugins"
+tags:
+- jmh
+- plugins
+- benchmark
+- performance
+- developer
+- gsoc
+- gsoc2019
+author: abhyudayasharma
+---
+
+= Performance Testing Framework for Jenkins Plugins
+
+I have been working on improving the performance of the Role Strategy Plugin as a part of my Google Summer of
+Code project. Since there was no existing way to measure performance and do benchmarks on Jenkins Plugins,
+my work for the first phase of this Google Summer of Code project was to create a framework for running
+benchmarks in Jenkins Plugins with a Jenkins instance available. To make our job a bit easier,
+we chose link:http://openjdk.java.net/jeps/230[Java Microbenchmark Harness] for running these benchmarks. This
+allows us to reliably measure performance of our time-critical functions and will help make Jenkins perform faster
+for everyone. The blogpost below shows how to run a benchmark in your plugin.
+
+== Creating Benchmarks
+
+The framework runs works by starting a temporary Jenkins instance for each fork of the JMH benchmark,
+just like `JenkinsRule` from Jenkins Test Harness. Benchmarks are run directly from your JUnit Tests which allows
+you to fail builds on the fly and easily run benchmarks from your IDE, just like unit tests. You can easily
+configure your benchmarks by either using your Java methods, or by using Jenkins Configuration-as-Code plugin
+and passing the path to your YAML file.
+
+To run benchmarks from your plugins, you either need to do the following:
+
+* bump up the minimum required Jenkins version to 2.60.3 or above
+* bump Plugin-POM to a version >= 3.46 or manually upgrade to Jenkins Test Harness >= 2.51.
+
+=== Writing benchmarks
+
+A reference to the Jenkins instance is available through either `JmhBenchmarkState#getJenkins()` or through
```

Comment:
Implementation detail instead of guidelines. Move code from `Sample benchmarks` here?